### PR TITLE
misc: prevent rewrite_shebang from erroring

### PR DESCRIPTION
**Pull request by branchv**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

---

Since https://github.com/Homebrew/brew/pull/12820, `rewrite_shebang` will error if none of the files given were rewritten. This was causing failures for some formula that called `rewrite_shebang` once per file. Instead, call it with all `bin` files.

### CODE REVIEW

File: Formula/glib.rb [hunk 1]
```diff
@@ -52,7 +52,7 @@ def install
       system "meson", *args, ".."
       system "ninja", "-v"
       system "ninja", "install", "-v"
-      bin.find { |f| rewrite_shebang detected_python_shebang, f }
+      rewrite_shebang detected_python_shebang, *Dir[bin/"*"]
```
Comment (carlocab) [inline Formula/glib.rb:55]:
```suggestion
      rewrite_shebang detected_python_shebang, *bin.children
```

**Comment by carlocab** [comment]
Also relevant: Homebrew/brew#12828

File: Formula/sql-translator.rb [hunk 2]
```diff
@@ -110,7 +110,7 @@ def install
     # Disable dynamic selection of perl which may cause segfault when an
     # incompatible perl is picked up.
     # https://github.com/Homebrew/homebrew-core/issues/4936
-    bin.find { |f| rewrite_shebang detected_perl_shebang, f }
+    rewrite_shebang detected_perl_shebang, *bin.children
```
Comment (carlocab) [inline Formula/sql-translator.rb:113]: You only changed one 😅

Comment (branchvincent) [reply Formula/sql-translator.rb:113]: ha oops, didn't save the rest of the files 💩

**Comment by carlocab** [comment]
```
  ==> Cleaning
  Still, the installation was successful, so we will link it into your prefix.
  Warning: The cleaning step did not complete successfully
```

Needs Homebrew/brew#12828

File: Formula/sql-translator.rb (hunk 2 above)
Comment (joshgoebel) [reply Formula/sql-translator.rb:113]: This is exactly what I was imagining might have needed to be done.  Looks good.  Thanks!

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**